### ISSUE TITLE
ebus registration on d-bus doesn't seem to happen right away

### ISSUE THREAD

**Issue by BenoitDuffez** [helium/miner#832]
We have one occurrence where the miner starts, does its stuff for 6 minutes and then only we have ebus connected to d-bus. There are no traces in the ebus module (we didn't add any either) so it's hard to tell, but it seems that it worked when we saw `STARTING UP MINER`. This line happened 6 minutes after `blockchain_sup:init:71 blockchain_sup init with ...`. 

Here are the logs: [console_dbus_issue.txt](https://github.com/helium/miner/files/6687191/console_dbus_issue.txt)

Let me know if I can provide more information.
tag @abhay thank you

**Comment by abhay** [comment]
Does this happen after a restart of the miner app or also on every restart of the device itself?

Any ideas @madninja? I'm suspecting that it has something to do with dbus name registration if it's happening only on miner app restarts.

**Comment by madninja** [comment]
The only time we've seen this is when miner is busy during startup. It happens.. We work around this by calling miner print_keys before starting miner so that external processes have access to the public key/name.

**Comment by BenoitDuffez** [comment]
This happened on the first start. On other starts/restarts it seems to be up faster. What's worrying is that it's a bit unpredictable.

@madninja I indeed rely on print_keys soon so this is not a problem. But we rely on ebus to build the transaction (CLI doesn't handle the txn/staking fees), so this could be problematic (we have a retry mechanism, but we don't want onboarding to be too slow).

**Comment by madninja** [comment]
> This happened on the first start. On other starts/restarts it seems to be up faster. What's worrying is that it's a bit unpredictable.

Unpredictable in my mind means that sometimes it never starts up.. you're saying it can take a few minutes to come up? That's not disastrous.. I'm thinking it's the snapshot loading which basically consumes the CPU. 

> @madninja I indeed rely on print_keys soon so this is not a problem. But we rely on ebus to build the transaction (CLI doesn't handle the txn/staking fees), so this could be problematic (we have a retry mechanism, but we don't want onboarding to be too slow).

Not sure we can do much more right now..

**Comment by BenoitDuffez** [comment]
I understand that it could lag a bit, but 6 minutes seems a bit much. Is 5 minutes within the expectation, even on a different hardware?

Is the solution just to wait & retry?

**Comment by madninja** [comment]
> I understand that it could lag a bit, but 6 minutes seems a bit much. Is 5 minutes within the expectation, even on a different hardware?
> 
> Is the solution just to wait & retry?

We'd need to see it in action? It takes longer on all hotspots I know.. The Kerlink cpu/memory is just not very speedy in comparison?

**Comment by BenoitDuffez** [comment]
Yes it's slower, but not 5 or 10 times slower. If the logs provided don't show the behavior let me know what could.
I could give you access to a gateway that runs it, but it's not predictable and doesn't happen all the time.

I wouldn't bother if it wasn't the system we use to build the transaction. The CLI doesn't support the txn fees so we can't use it, otherwise it would perhaps be a better candidate (slower than dbus, but more reliable). Maybe that would be the solution? (add support for txn fees in CLI `add_gateway`)

**Comment by madninja** [comment]
> Yes it's slower, but not 5 or 10 times slower. If the logs provided don't show the behavior let me know what could.
> I could give you access to a gateway that runs it, but it's not predictable and doesn't happen all the time.
> 
> I wouldn't bother if it wasn't the system we use to build the transaction. The CLI doesn't support the txn fees so we can't use it, otherwise it would perhaps be a better candidate (slower than dbus, but more reliable). Maybe that would be the solution? (add support for txn fees in CLI `add_gateway`)

it's certainly possible to add the txn fee as an option to the miner CLI. The issue is that the cli still has to talk to the miner and if it's too busy for bus, is it also too busy for cli rpc work?
